### DISCUSSION TITLE
Opening context menu makes the tabcontrol items datacontext null

### DISCUSSION THREAD

**Discussion by qqdown** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2929] [Q&A] [answered]
When I created a TabControl with the original control style and right-clicked to open a context menu, everything works fine.

However, when I applied the MaterialDesigin theme and did the same operation, **the DataContext of TabControl's items become null.** 

The startup view:

![image](https://user-images.githubusercontent.com/13580690/199198356-8ece06a2-2f89-4b1b-a7cb-5dba0157e6c6.png)

Then **I right-clicked the textbox and opened the context menu, the tab item contents were gone:**

![image](https://user-images.githubusercontent.com/13580690/199198421-8151f9a5-e892-45a1-b565-e6e9ea53c116.png)

From the visual hierarchy view, I found **the DataContext of the TextBox was changed from ViewModel to null**. If I add a DataContextChanged event to the TextBox, the event will be fired.

![image](https://user-images.githubusercontent.com/13580690/199199793-d24e6351-7646-494e-b47b-195e501456a5.png)

Before right click:

![image](https://user-images.githubusercontent.com/13580690/199199812-7382e7f0-5853-4263-9f77-3162b04d2459.png)

After right click:

![image](https://user-images.githubusercontent.com/13580690/199199860-c864a2ee-b0cc-4657-8465-2da4ca72a487.png)

The DataContext become null and the TextBox's content went empty, but why DataCotext changed?

Is there something I did incorrectly?

Here is the code:

MainWindows.xaml.cs

```
 public class ViewModel : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;

        private string _Name;
        public string Name
        {
            get => _Name; set { _Name = value; OnPropertyChanged("Name"); }
        }

        private void OnPropertyChanged(string name)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
        }
    }

    public partial class MainWindow : Window
    {
        public ObservableCollection<ViewModel> ViewModels { get; set; } = new ObservableCollection<ViewModel>();

        public MainWindow()
        {
            InitializeComponent();
            ViewModels.Add(new ViewModel() { Name = "aaaa" });
            ViewModels.Add(new ViewModel() { Name = "bbbb" });
            DataContext = this;
        }
    }
```

MainWindow.cs

```
<TabControl ItemsSource="{Binding ViewModels}" SelectedIndex="0">
            <TabControl.ItemTemplate>
                <DataTemplate>
                    <TextBlock Text="{Binding Name}"/>
                </DataTemplate>
            </TabControl.ItemTemplate>
            <TabControl.ContentTemplate>
                <DataTemplate>
                    <TextBox Text="{Binding Name, UpdateSourceTrigger=PropertyChanged}"/>
                </DataTemplate>
            </TabControl.ContentTemplate>
        </TabControl>
```

**Comment by nicolaihenriksen** [comment]
@qqdown I just had a brief look at this. I can quite easily reproduce it in the demo app, and something seems to be really strange. I need to dig a bit deeper, but it seems if you set `HorizontalContentAlignment="Left"` (or `Right`) on the `TabControl`, the issue seems to disappear.

The issue seems to occur when a new "window" opens; this can either be a `ToolTip` or a `ContextMenu`.
Shown below in the demo tool (with `HorizontalContentAlignment` temporarily removed), where the content disappears after simply hovering an element with a `ToolTip`:

![DataContextDisappears](https://user-images.githubusercontent.com/19572699/204327706-b96f41ee-045f-425a-9edf-721ed4696f63.gif)

**Reply by nicolaihenriksen** [reply]
@qqdown I created an issue for this. It is easily reproducible, even with a UI test, so this is something that needs to be fixed I believe. The issue is here: #2983

**Reply by nicolaihenriksen** [reply, marked as answer]
@qqdown The issue has been fixed. You can grab one of the nightly builds and verify it if you want.
